PPTT-2715 added links, improved some content (#681)
PiwikPRO/PPMS-PublicDocs: data_collection/web/guides.rst

.. note::

    Automatic content tracking can be enabled in Tag Manager, as shown in
    `Set up content tracking <https://help.piwik.pro/support/questions/set-up-content-tracking/>`_
    article.

**But how does JavaScript Tracking Client know what blocks you would like to track?**

There are two ways of marking HTML elements as content blocks: you must either add a special attribute ``data-track-content`` or class ``piwikTrackContent``. Example:

.. code-block:: html
    :linenos:

    <a href="http://example.com/image/abc.png" title="abc" data-track-content>
        first content block
    </a>
    <a href="http://example.com/image/def.png" title="def" class="piwikTrackContent">
        second content block
    </a>

Content properties will be taken from HTML attributes of the content block element or any of its descendants:

* name comes from ``data-content-name`` attribute
* piece comes from ``data-content-piece`` attribute
* target comes from ``data-content-target`` attribute

If any of these attributes is missing, JavaScript Tracking Client will try extracting the value from other sources, using the following logic:

> * piece will be taken from ``src`` attribute of an element with ``piwikContentPiece`` class or block element

Author: You're right. It was hard to fit this information into the guide though. I think `src` is an acceptable simplification.

> * piece will be taken from ``src`` attribute of an element with ``piwikContentPiece`` class or block element

Reviewer: Yes, we can leave it as is or just mention that video is also processed in few ways, but without specifics.

* target will be taken from ``href`` attribute of an element with ``piwikContentTarget`` class, block element or piece element

> * name will be a copy of piece value if present, or taken from ``title`` attribute of block element, piece element or target element

Reviewer: ```suggestion
* name will try to use piece value if present, otherwise it'll be taken from ``title`` attribute of block element, piece element or target element
```

And goal tracking is about tracking conversions. If you can point out parts of your website/application more important from your business perspective, you could :ref:`define those parts as goals<https://help.piwik.pro/support/analytics-new/goals/>`.
Visiting a specific landing page, submitting a contact form, downloading a PDF file with your product manual - these are popular examples of goal definitions. You can even define a goal based on the custom event you are tracking.

If a goal with automatic tracking is defined in Analytics, every time an events matching the goal's definition is tracked, we create an additional conversion event and save it along the original event. We call this procedure an "automatic conversion".

.. note::

    `Set up goals <https://help.piwik.pro/support/analytics-new/goals/>`_ article shows how to define a goal triggered by visiting a specific page.

Alternatively, you can trigger a goal manually with the use of :ref:`trackGoal<jtc-api-trackGoal>` function

.. code-block:: javascript

    // force conversion of the goal with ID 17
    _paq.push(["trackGoal", 17]);

> before triggering :ref:`trackPageView<jtc-api-trackPageView>`.
> 
> We call this procedure a "manual conversion". Manual conversion doesn't create an additional conversion event like the automatic conversion does.

Reviewer: ```suggestion
We call this procedure a "manual conversion". Manual conversion will sent a new standalone event immediately and is not tied to any other events sent by tracker like automatic conversions.
```
I tried to say that goal event is sent immediately when this command is processed (without waiting on page view).
